__tests__/Unit/Components/Tasks/TaskDetails.test.tsx

```diff
@@ -659,3 +643,78 @@ describe('TaskDependency', () => {
         });
     });
 });
+
+jest.mock('next/router', () => ({
+    useRouter() {
+        return {
+            query: {
+                dev: 'true',
+            },
+            push: mockNavigateToUpdateProgressPage,
+        };
+    },
+}));
+
+describe('Details component', () => {
+    it('Renders the detail type and value correctly', () => {
+        renderWithRouter(<Details detailType="Assignee" value="John Doe" />);
+        const detailType = screen.getByText('Assignee:');
+        const detailValue = screen.getByText('John Doe');
+        expect(detailType).toBeInTheDocument();
+        expect(detailValue).toBeInTheDocument();
+    });
+
+    it('Renders a GitHub link correctly', () => {
+        const githubIssueUrl = 'https://github.com/example/repo/issues/123';
+        renderWithRouter(<Details detailType="Link" value={githubIssueUrl} />);
+        const link = screen.getByRole('link', { name: /Open GitHub Issue/i });
+        expect(link).toBeInTheDocument();
+        expect(link).toHaveAttribute('href', githubIssueUrl);
+    });
+
+    it('Does not display tooltip on hover when timestamp is not provided', async () => {
+        render(<Details detailType="Started On" value="N/A" />);
+        const detailValue = screen.getByText('Started:');
+        fireEvent.mouseOver(detailValue);
+        const tooltip = screen.queryByText('N/A', { selector: '.tooltip' });
+        expect(tooltip).not.toBeInTheDocument();
+    });
+
+    it('Renders an extension request icon for Ends On with URL', () => {
+        const url = 'https://example.com';
+        renderWithRouter(<Details detailType="Ends On" url={url} />);
+        const extensionRequestIcon = screen.getByTestId(
+            'extension-request-icon'
+        );
+        expect(extensionRequestIcon).toBeInTheDocument();
+        expect(extensionRequestIcon).toHaveAttribute('href', url);
+    });
+
+    it('Does not render an extension request icon for Ends On without URL', () => {
+        renderWithRouter(<Details detailType="Ends On" />);
+        const extensionRequestIcon = screen.queryByTestId(
+            'extension-request-icon'
+        );
+        expect(extensionRequestIcon).not.toBeInTheDocument();
+    });
+    it('Renders a tooltip for timestamps', async () => {
+        const task = {
+            id: 'L1SDW6O835o0EI8ZmvRc',
+            startedOn: '1707869428.523',
+        };
+        renderWithRouter(
+            <Details detailType="Started On" value={task.startedOn} />
+        );
+        try {
+            const detailTypeElement = screen.getByText('Started:');
+            expect(detailTypeElement).toBeInTheDocument();
+            fireEvent.mouseOver(detailTypeElement);
+            const tooltip = await screen.findByTestId('tooltip');
```

Comment:
why await findByTestId?

Reply:
The `findByTestId` function is used to asynchronously find an element in the rendered component based on its test ID. Since the rendering of the tooltip is an asynchronous operation, we need to wait for it to appear in the DOM before making assertions about its presence. Using `findByTestId` allows us to await the appearance of the tooltip before proceeding with the test assertions. This ensures that the test waits for the tooltip to be rendered before making any assertions about its presence and content.